test/testsuites/opsrcdeletetests.go

```diff
@@ -0,0 +1,56 @@
+package testsuites
+
+import (
+	"testing"
+
+	"github.com/operator-framework/operator-marketplace/test/helpers"
+	"github.com/operator-framework/operator-sdk/pkg/test"
+)
+
+// DeleteOpSrc tests that the correct cleanup occurs when an OpSrc is deleted
+func DeleteOpSrc(t *testing.T) {
+	t.Run("delete-operator-source", testDeleteOpSrc)
+}
+
+// testDeleteOpSrc ensures that after deleting an OperatorSource that the
+// objects created as a result are deleted
+func testDeleteOpSrc(t *testing.T) {
+	ctx := test.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	// Get global framework variables
+	client := test.Global.Client
+
+	// Get test namespace
+	namespace, err := test.NewTestCtx(t).GetNamespace()
+	if err != nil {
+		t.Errorf("Could not get namespace: %v", err)
+	}
+
+	testOperatorSource := helpers.CreateOperatorSourceDefinition(namespace)
+
+	// Create the OperatorSource with no cleanup options.
+	err = helpers.CreateRuntimeObjectNoCleanup(client, testOperatorSource)
+	if err != nil {
+		t.Errorf("Could not create operator source: %v", err)
```

Comment:
Please use `assert` given that we are going that route now. Comment applies to all instances.